p2p/nat/nat.go

```diff
@@ -243,7 +247,13 @@ func (nat *NAT) addMapping(m *mapping) {
 	nat.proc.AddChild(m.proc)
 
 	nat.mappingmu.Lock()
-	nat.mappings = append(nat.mappings, m)
+	nat.mappings[m] = struct{}{}
+	nat.mappingmu.Unlock()
+}
+
+func (nat *NAT) rmMapping(m *mapping) {
```

Comment:
Yeah... I feel like our highest priority right now should be the addrsplosion issue.